block/manager.go

```diff
@@ -346,17 +371,28 @@ func (m *Manager) AggregationLoop(ctx context.Context, lazy bool) {
 			// building a block.
 			case _, ok := <-m.txsAvailable:
 				if ok && !m.buildingBlock {
+					// set the buildingBlock flag to prevent multiple calls to reset the timer
 					m.buildingBlock = true
-					timer.Reset(1 * time.Second)
-				}
-			case <-timer.C:
-				// build a block with all the transactions received in the last 1 second
-				err := m.publishBlock(ctx)
-				if err != nil && ctx.Err() == nil {
-					m.logger.Error("error while publishing block", "error", err)
+					// Reset the block timer based on the block time and the default sleep.
+					blockTimer.Reset(getRemainingSleep(start, m.conf.BlockTime, defaultSleep))
```

Comment:
is the intention here is to wait for arbitrary amount of time? for instance, 
* if start is 0, you wait for `defaultSleep`
* if (now-start) is close to BlockTime, you may wait < defaultSleep
* if (now-start) is > defaultSleep, but < BlockTime, you may wait for > defaultSleep?

Reply:
Updated comments for`getRemainingSleep`.

https://github.com/rollkit/rollkit/pull/1643/files#diff-0a427fe6ee00070e86bdb14930af6ae758cf682d702e12a2f386c56f20a5de2bR730

lmk if that clarifies things. 

The updates to `getRemainingSleep` were to generalize it for the 3 different sleep duration calculations.
1. Coming out of lazy mode.
2. Determining when to produce the next block based on the lazyblocktime
3. Determining when to produce the next block based on the blocktime

Reply:
update comments in code here as well.